Allow configuration of the GC log file via an environment variable

ankon opened the pull request:
Enabling GC logging works now by setting the environment variable ES_GC_LOG_FILE
to the full path to the GC log file. Missing directories will be created as needed.

The ES_USE_GC_LOGGING environment variable is no longer used.

Closes #8471
Closes #8479

Note that the default name of the file was already proper for Windows, so the fix for #8471 effectively is "do what windows did".

ankon committed:
Allow configuration of the GC log file via an environment variable

Enabling GC logging works now by setting the environment variable ES_GC_LOG_FILE
to the full path to the GC log file. Missing directories will be created as needed.

The ES_USE_GC_LOGGING environment variable is no longer used.

Closes #8471

ankon commented:
Anything on my side to be done to get this merged?

spinscale commented:
Hey,

looks good. Can you rebase & squash and then its ready from my point of view.
Sorry, this one slipped under my rader for too long.

ankon commented:
Hi, I squashed them together on top of the (then-current :D) master.

While doing that I fixed a minor issue for Windows: it didn't create the directories. This was tested with a Windows 10 cmd.exe, and worked as expected. Note that while the bat code handles them, the JVM does not actually allow paths with spaces for the GC log.

spinscale commented:
just merged it in, thx a lot for helping and your patience!
